iot_verifier/src/gateway_updater.rs

```diff
@@ -0,0 +1,90 @@
+use crate::Settings;
+use chrono::Duration;
+use futures::stream::StreamExt;
+use helium_crypto::PublicKeyBinary;
+use iot_config::{
+    client::{Client as IotConfigClient, ClientError as IotConfigClientError},
+    gateway_info::{GatewayInfo, GatewayInfoResolver},
+};
+use std::collections::HashMap;
+use tokio::sync::watch;
+use tokio::time;
+
+pub type GatewayMap = HashMap<PublicKeyBinary, GatewayInfo>;
+pub type MessageSender = watch::Sender<GatewayMap>;
+pub type MessageReceiver = watch::Receiver<GatewayMap>;
+
+pub struct GatewayUpdater {
+    iot_config_client: IotConfigClient,
+    refresh_interval: Duration,
+    sender: MessageSender,
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum GatewayUpdaterError {
+    #[error("gateway not found: {0}")]
+    GatewayNotFound(PublicKeyBinary),
```

Comment:
is this a valid error now since the updater is never looking for any specific gateway? perhaps this should be replaced by the SendError per the comment below?

Reply:
addressed here: fdabbdfa32b12cc10f890d753fa15726fd7effd8